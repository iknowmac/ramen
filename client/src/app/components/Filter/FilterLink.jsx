'use strict';
import React, { PropTypes } from 'react';

const FilterLink = (props) => {

  const handleClick = (e) => {
    e.preventDefault();
    props.onClick();
  };

  if (props.isActive) {
    return <span className="filter__button filter__button--active">{props.children}</span>;
  }

  return <a href="#" className="filter__button" onClick={handleClick}>{props.children}</a>;

};

FilterLink.propTypes = {
  isActive: PropTypes.bool.isRequired,
  children: PropTypes.node.isRequired,
  onClick: PropTypes.func.isRequired
}

export default FilterLink;
