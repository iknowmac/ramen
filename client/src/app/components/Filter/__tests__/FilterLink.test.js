'use strict';
import React from 'react';
import FilterLink from '../FilterLink';
import renderer from 'react-test-renderer';

describe('FilterLink', () => {

    const mockEvent = new Event("onClick");
    mockEvent.preventDefault = (e) => {}

    const mockHandler = () => {}

    it('renders correctly', () => {
        const tree = renderer.create(
            <FilterLink onClick={mockHandler} isActive={false}>All</FilterLink>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders as a span when isActive is true', () => {
        const tree = renderer.create(
            <FilterLink onClick={mockHandler} isActive={true}>All</FilterLink>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders as with active css class when isActive is true', () => {
        const tree = renderer.create(
            <FilterLink onClick={mockHandler} isActive={true}>All</FilterLink>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('fires event for supplied onClick handler', () => {
        const component = renderer.create(
            <FilterLink onClick={mockHandler} isActive={false}>All</FilterLink>
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        // manually trigger the callback
        tree.props.onClick(mockEvent);
        // re-rendering
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();

    });

});