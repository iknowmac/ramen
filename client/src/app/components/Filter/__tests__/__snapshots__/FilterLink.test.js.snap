exports[`FilterLink fires event for supplied onClick handler 1`] = `
<a
  className="filter__button"
  href="#"
  onClick={[Function]}>
  All
</a>
`;

exports[`FilterLink fires event for supplied onClick handler 2`] = `
<a
  className="filter__button"
  href="#"
  onClick={[Function]}>
  All
</a>
`;

exports[`FilterLink renders as a span when isActive is true 1`] = `
<span
  className="filter__button filter__button--active">
  All
</span>
`;

exports[`FilterLink renders as with active css class when isActive is true 1`] = `
<span
  className="filter__button filter__button--active">
  All
</span>
`;

exports[`FilterLink renders correctly 1`] = `
<a
  className="filter__button"
  href="#"
  onClick={[Function]}>
  All
</a>
`;
